examples/plotting/wcsaxes_plotting_example.py

```diff
@@ -0,0 +1,51 @@
+"""
+=====================================
+Plotting line on a Map with WCSAxes
+=====================================
+Example to demonstrate plotting of point, line and arch in
+pixel coordinate, world coordinate and sky coordinate respectively.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.data.sample
+import sunpy.map
+from sunpy.coordinates.utils import GreatArc
+
+###############################################################################
+# we start with the sample data.
+my_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+fig = plt.figure()
+ax = fig.add_subplot(projection=my_map)
+my_map.plot(axes=ax, clip_interval=(1, 99.9)*u.percent)
+
+# plot a line on map origin from center using WCSAxes
+xx = np.arange(0, 500)
+yy = xx
+ax.plot(xx*u.arcsec.to(u.deg), yy*u.arcsec.to(u.deg),
+        color='r',
+        transform=ax.get_transform("world"),
+        label=f'WCS coordinate [{0*u.arcsec}, {500*u.arcsec}]')
+
+# plot a point in middle of the image using pixel coordinate
+pixel_coord = [my_map.data.shape[0]/2., my_map.data.shape[1]/2.] * u.pix
+ax.plot(pixel_coord[0], pixel_coord[1], 'x', color='k',
+        label=f'Pixel coordinate [{pixel_coord[0]}, {pixel_coord[1]}]')
+ax.coords.grid(color='yellow', linestyle='solid', alpha=0.5)
+
```

Comment:
I would add one more example showing just how to plot a one point using a SkyCoord (just to highlight the difference between plotting world coordinates using the `transform` method, plotting in pixel space and then plotting with `.plot_coord` with SkyCoord.

Reply:
> I've left some suggestions regarding being more descriptive in what is being done here as it is a gallery example.

Hey @hayesla  I have made the suggested changes. Please have a look : )

> It would also be good just to add one more example here to show how to plot a simple point with SkyCoord. 

Does one more example means adding another example? one more commit or PR?

> how to plot a one point using a SkyCoord (just to highlight the difference between plotting world coordinates using the transform method, plotting in pixel space and then plotting with .plot_coord with SkyCoord.

I was thinking if you could elaborate a bit further. We have to plot three **point** here in three different **coordinates** using the **methods** you specified. Am i thinking clearly?

Reply:
> That looks correct to me on first glance. Maybe move the point elsewhere (I assume it will be on the same image as the other shapes).

Hey Nabil! How about this! please let me know if anything need fixing!
I am unable to move label upside as it's default wherever it find space it pop up :smile: ! does this require our attention?

![Skycoords](https://user-images.githubusercontent.com/72349558/109105552-2ebc9080-7754-11eb-99cc-e81ef2b21136.png)

Reply:
We should fix the legends location instead of letting it move around.

Reply:
you are right! i just did it! i decreased the figure size to 8,8 as well!
is this better?
![wcs](https://user-images.githubusercontent.com/72349558/109132616-0b0c4100-777a-11eb-8d6d-e67097d8f1fd.png)

Reply:
Works for me.